dashboard/src/actions/apps.ts

```diff
@@ -86,6 +146,18 @@ export function fetchApps(
     try {
       const apps = await App.listApps(ns, all);
       dispatch(receiveAppList(apps));
+      const appsWithUpdateInfo = await Promise.all(
+        apps.map(
+          async (app): Promise<IAppOverviewWithUpdateInfo> => {
+            const name = app.chartMetadata.name;
+            const currentVersion = app.chartMetadata.version;
+            const appVersion = app.chartMetadata.appVersion;
+            const updateInfo = await getChartUpdates(name, currentVersion, appVersion);
```

Comment:
It's a much better idea to have `getChartUpdates` be an action that we simply dispatch and no longer care about. The `getChartUpdates` action can then be responsible for updating the entry in the `IAppState["listOverview"]` and `IAppState["selected"]` if it matches (we may want to change listOverview to the byID pattern to make it easier to reference). This way, we don't need to create a bunch of Promises and await on `Promise.all`.

This is important, because it will give us a separate Redux action for each returned app update rather than using the same one (e.g. we will get a `RECEIVE_APP_UPDATE_INFO` per app update retrieved). It's also not performant to update the whole array in one go, rather than items in the array. This will likely force a React render all the way up the tree. Also, `getChartUpdates` might need to control it's own lifecycle, e.g. if we needed a separate `isFetching` for the updatesInfo, we would want to dispatch the states from there.

We might want to do this in a separate PR, after this is merged, given there might be a lot of changes if we need to change this to the byID pattern.

Reply:
I get your point. I think this only applies to `IAppState["listOverview"]` since when getting a single app we do it one by one. Since we already have an issue for refactoring the byID schedule.

If we avoid making `getChartUpdates` an action, the changes to update info per app would be minimal and will fit in this PR.